core/src/consensus.rs

```diff
@@ -1027,7 +1029,13 @@ impl Tower {
             retain_flags_for_each_vote_in_reverse.into_iter().rev();
 
         let original_votes_len = self.lockouts.votes.len();
-        self.initialize_lockouts(move |_| retain_flags_for_each_vote.next().unwrap());
+        let keep_the_only_root_vote_for_no_double_vote =
```

Comment:
comment here why this must be accepted.

Reply:
because we start to vote on root....

also, it's very very rare to see this on the wild. I've just come up with this possibility while working on #12671.

Reply:
We can also augment the check here https://github.com/solana-labs/solana/blob/master/core/src/consensus.rs#L454 to check that the vote slot > root so the `is_locked_out` check will fail